plugins/modules/event_info.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Paul Arthur <paul.arthur@flowerysong.com>
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'XLAB Steampunk'}
+
+DOCUMENTATION = '''
+module: event_info
+author:
+  - Paul Arthur (@flowerysong)
+  - Aljaz Kosir (@aljazkosir)
+  - Manca Bizjak (@mancabizjak)
+  - Tadej Borovsak (@tadeboro)
+short_description: Lists Sensu events
+description:
+  - For more information, refer to the Sensu documentation at
+    U(https://docs.sensu.io/sensu-go/latest/reference/events/)
+extends_documentation_fragment:
+  - sensu.sensu_go.base
+options:
+  check_name:
+    description:
+      - Limit results to a specific check.
+      - C(entity_name) must also be specified.
+    type: str
+  entity_name:
+    description:
+      - Limit results to a specific entity.
+    type: str
+'''
+
+EXAMPLES = '''
+- name: List Sensu events
+  event_info:
+  register: result
+
+- name: List Sensu events for api.example.com
+  event_info:
+    entity: api.example.com
+  register: result
+'''
+
+RETURN = '''
+objects:
+  description: list of Sensu events
+  returned: always
+  type: list
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    arguments, errors, utils,
+)
+
+
+def main():
+    required_by = {'check_name': ['entity_name']}
+    module = AnsibleModule(
+        supports_check_mode=True,
+        required_by=required_by,
+        argument_spec=dict(
+            arguments.get_spec("auth"),
+            check_name=dict(),
+            entity_name=dict(),
```

Comment:
Despite reading the docs at the beginning of this file first,  the example left me a bit confused. I had to stop and think for a second about what `entity_name` and `check_name` are. Somehow I felt like we're actually trying to configure something (not filters!), even though I know `*_info` modules never change the state of resources on the backend.

If I were the user of the module, I think I'd prefer writing a task like this:
```yaml
- name: List Sensu events for api.example.com
  event_info:
    filters:
     - entity: api.example.com  # or entity_name
     - check: some-check        # or check_name
  register: result
```

Grouping supported filters under optional `filters` parameter seems a bit more intuitive to me, and I feel like it expresses our intentions better. But maybe not all end-users are as confused as me ;)

What do you guys think? @tadeboro

Reply:
On the one hand, I kind of like the `filters` approach. It is explicit about what we are doing. And we can quickly transform it into something that can retrieve events from different sources. I would add one little change to the proposed API. Because the `check` parameter is dependent on `entity` being present, I would group them like this:

    - event_info:
        filters:
         - entity: first_enity
           check: first_check
         - entity: second_entity
         - entity: third_enity
           check: third_check
      register: result

In a way, the `entity` and `check` pair is a single filter.

But on the other hand, if we stick with a top-level `entity` and `check` parameters, we can retain the symmetry that all other module pairs have. For example, we create an asset like this:

    - asset:
        name: my_asset
        # other params here

If we want to retrieve the information about the asset we created, we add an `_info` suffix and drop everything except the name:

    - asset_info:
        name: my_asset

Now, the event module's API will look something like this (@aljazkosir, please correct me if I am wrong):

    - event:
        entity: my_entity
        check: my_check
        # other params here

`entity` and `check` parameter pair has a very similar function as the `name` parameter in other modules. It feels natural that we query about the events in roughly the same way:

    - event_info:
        entity: my_entity
        check: my_check

Now, I know that events are a bit special in that there can be more than one instance that matches the `entity` and `check` parameters, but the API still feels kind of right.

Again, I am fine with any of the approaches here. Maybe I am a bit biased towards the second version (where the `entity` and `check` are not nested under `filters`) because it feels it should be less code to implement. And less code is good ;) But if you both vote for the other approach, we will go with that.